src/main/java/jenkins/plugins/office365connector/Webhook.java

```diff
@@ -106,22 +104,101 @@ public boolean isNotifyRepeatedFailure() {
     }
 
     public int getTimeout() {
-            return timeout;
+        return timeout == 0 ? DEFAULT_TIMEOUT : timeout;
     }
 
     public List<Macro> getMacros() {
         return macros;
     }
 
-    public FormValidation doCheckURL(@QueryParameter(value = "url", fixEmpty = true) String url) {
-        if (url.equals("111"))
-            return FormValidation.ok();
-        else
-            return FormValidation.error("There's a problem here");
+    @DataBoundSetter
+    public void setTimeout(int timeout) {
+        this.timeout = timeout;
+    }
+
+    @DataBoundSetter
+    public void setName(String name) {
+        this.name = Util.fixEmptyAndTrim(name);
+    }
+
+    @DataBoundSetter
+    public void setStartNotification(boolean startNotification) {
+        this.startNotification = startNotification;
+    }
+
+    @DataBoundSetter
+    public void setNotifySuccess(boolean notifySuccess) {
+        this.notifySuccess = notifySuccess;
+    }
+
+    @DataBoundSetter
+    public void setNotifyAborted(boolean notifyAborted) {
+        this.notifyAborted = notifyAborted;
+    }
+
+    @DataBoundSetter
+    public void setNotifyNotBuilt(boolean notifyNotBuilt) {
+        this.notifyNotBuilt = notifyNotBuilt;
+    }
+
+    @DataBoundSetter
+    public void setNotifyUnstable(boolean notifyUnstable) {
+        this.notifyUnstable = notifyUnstable;
+    }
+
+    @DataBoundSetter
+    public void setNotifyFailure(boolean notifyFailure) {
+        this.notifyFailure = notifyFailure;
+    }
+
+    @DataBoundSetter
+    public void setNotifyBackToNormal(boolean notifyBackToNormal) {
+        this.notifyBackToNormal = notifyBackToNormal;
+    }
+
+    @DataBoundSetter
+    public void setNotifyRepeatedFailure(boolean notifyRepeatedFailure) {
+        this.notifyRepeatedFailure = notifyRepeatedFailure;
+    }
+
+    @DataBoundSetter
+    public void setMacros(List<Macro> macros) {
+        this.macros = Util.fixNull(macros);
     }
 
     @Override
     public String toString() {
         return url;
     }
-}
\ No newline at end of file
+
+    @Extension
+    public static class DescriptorImpl extends Descriptor<Webhook> {
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return "Webhook";
+        }
+
+        public int getDefaultTimeout() {
+            return Webhook.DEFAULT_TIMEOUT;
+        }
+
+        public FormValidation doCheckUrl(@QueryParameter String value) {
+            if (StringUtils.isBlank(value)) {
+                return FormValidation.error("URL or variable must be provided");
+            }
+            if (value.startsWith("$") && value.length() > 1) {
+                return FormValidation.ok();
+            }
+
+            try {
+                new URL(value);
+            } catch (MalformedURLException e) {
```

Comment:
looks like code which can be exposed to separate method or merged
https://github.com/jenkinsci/office-365-connector-plugin/pull/24/files#diff-c8d3ccb3e6896886dafa3f9232f4f15cR142

Reply:
I agree, and considered it 👍 Just didn't act on it.